pkg/driver/file/device_manager_file.go

```diff
@@ -732,6 +733,49 @@ func (d *DeviceManager) WriteMetrics(u uuid.UUID, b []byte) error {
 	return dev.AddMetrics(b)
 }
 
+// WriteCerts write an attestation certs information
+func (d *DeviceManager) WriteCerts(u uuid.UUID, b []byte) error {
+	// refresh certs from filesystem, if needed - includes checking if necessary based on timer
+	err := d.refreshCache()
+	if err != nil {
+		return fmt.Errorf("unable to refresh certs from filesystem: %v", err)
+	}
+	// look up the device by uuid
+	_, ok := d.devices[u]
+	if !ok {
+		return fmt.Errorf("unregistered device UUID %s", u.String())
+	}
+	if len(b) < 1 {
+		return fmt.Errorf("empty configuration")
+	}
+	// save the base configuration
+	err = d.writeJSONFile(u, "", deviceAttestCertsFilename, b)
+	if err != nil {
+		return fmt.Errorf("error saving attestation to %s: %v", deviceAttestCertsFilename, err)
+	}
+	return nil
+}
+
+// GetCerts retrieve the attest certs for a particular device
+func (d *DeviceManager) GetCerts(u uuid.UUID) ([]byte, error) {
+	// read the config from disk
+	fullAttestPath := path.Join(d.getDevicePath(u), deviceAttestCertsFilename)
+	b, err := ioutil.ReadFile(fullAttestPath)
+	switch {
+	case err != nil && os.IsNotExist(err):
+		// create the base file if it does not exist
+		b = common.CreateBaseConfig(u)
+		err = d.writeJSONFile(u, "", deviceAttestCertsFilename, b)
+		if err != nil {
+			return nil, fmt.Errorf("error saving device config to %s: %v", deviceAttestCertsFilename, err)
```

Comment:
Same with `path.Join()`? Also this should be an error "_reading_  device config from ... "

Reply:
It was unneeded block here. Removed